docs/docsite/rst/dev_guide/testing_running_locally.rst

```diff
@@ -2,44 +2,329 @@
 
 .. _testing_running_locally:
 
-***************
-Testing Ansible
-***************
+*******************************
+Testing Ansible and Collections
+*******************************
 
-This document describes how to:
-
-* Run tests locally using ``ansible-test``
-* Extend
+This document describes how to run tests using ``ansible-test``.
 
 .. contents::
    :local:
 
-Requirements
-============
+Setup
+=====
 
-There are no special requirements for running ``ansible-test`` on Python 2.7 or later.
-The ``argparse`` package is required for Python 2.6.
-The requirements for each ``ansible-test`` command are covered later.
+Before running ``ansible-test``, set up your environment for :ref:`Testing an Ansible Collection` or
+:ref:`Testing ansible-core`, depending on which scenario applies to you.
 
+.. warning::
 
-Test Environments
-=================
+   If you use ``git`` for version control, make sure the files you are working with are not ignored by ``git``.
+   If they are, ``ansible-test`` will ignore them as well.
+
+Testing an Ansible Collection
+-----------------------------
+
+If you are testing an Ansible Collection, you need a copy of the collection, preferably a git clone.
+For example, to work with the ``community.windows`` collection, follow these steps:
+
+1. Clone the collection you want to test into a valid collection root:
+
+   .. code-block:: shell-session
```

Comment:
`shell-session` is the same as `console` — it expects to highlight terminal capture, meaning both commands and their outputs. The commands are expected to begin with a $-prompt — this is how this lexer recognizes where it's fine to highlight shell syntax and there it's not (it doesn't attempt to highlight the output).
```suggestion
   .. code-block:: shell
```